Mathlib/Topology/Algebra/Module/WeakBilin.lean

```diff
@@ -136,6 +138,19 @@ instance instContinuousSMul [ContinuousSMul 𝕜 𝕜] : ContinuousSMul 𝕜 (We
   simp only [Function.comp_apply, Pi.smul_apply, LinearMap.map_smulₛₗ, RingHom.id_apply,
     LinearMap.smul_apply]
 
+/--
+Map F into the topological dual of E with the weak topology induced by F
+-/
+def _root_.LinearMap.dualPairing [ContinuousAdd 𝕜] [ContinuousConstSMul 𝕜 𝕜] :
+    F →ₗ[𝕜] (WeakBilin B) →L[𝕜] 𝕜 where
+  toFun := fun x => ⟨B.flip x, WeakBilin.eval_continuous _ _⟩
+  map_add' := fun x y => by
+    simp only [map_add]
+    rfl
+  map_smul' := fun r x => by
+    simp only [map_smul, RingHom.id_apply]
+    rfl
```

Comment:
The `fun_prop` call works if you mark `WeakBilin.eval_continuous` with it.
```suggestion
/--
Map `F` into the topological dual of `E` with the weak topology induced by `F`
-/
def eval [ContinuousAdd 𝕜] [ContinuousConstSMul 𝕜 𝕜] :
    F →ₗ[𝕜] WeakBilin B →L[𝕜] 𝕜 where
  toFun f := ⟨B.flip f, by fun_prop⟩
  map_add' _ _ := by ext; simp
  map_smul' _ _ := by ext; simp
```
I'm open to pushback on the name, but I think this matches, for example, [Module.Dual.eval](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Dual.html#Module.Dual.eval).

Reply:
@j-loreaux I got the name `dualPairing` from [here](https://github.com/leanprover-community/mathlib4/blob/d46cea728d1e4077b4ec97f62ec0796057a07fd3/Mathlib/Topology/Algebra/Module/WeakDual.lean#L60) but I don't have a strong preference for it.